examples/signature-detection/langgraph-agent-signature-detection-cli/signature_detection_langgraph_agent.py

```diff
@@ -215,164 +55,94 @@ class SignatureConversationAgent:
-            print(f"Error: {e}")
+I've processed a document and got this result:
+{parsed_result}
+
+Please analyze the above parsed output and answer the following:
+{question_block}
+"""
+    return system_prompt
+
+
+# Define the agent state
+class AgentState(TypedDict):
+    messages: Annotated[list, add_messages]
+
+
+# Agent node - decides whether to use tools
+async def agent_node(state: AgentState):
+    model = ChatOpenAI(model="gpt-4o", temperature=0.1).bind_tools(
+        [document_markdown_tool]
+    )
+
+    response = await model.ainvoke(state["messages"])
+    return {"messages": [response]}
+
+
+# Conditional Logic for Tool Use
+def should_continue(state: AgentState):
+    last_message = state["messages"][-1]
+    if hasattr(last_message, "tool_calls") and last_message.tool_calls:
+        return "tools"
+    return END
+
 
+# LangGraph Workflow Setup
+workflow = StateGraph(AgentState)
+workflow.add_node("agent", agent_node)
+workflow.add_node("tools", ToolNode([document_markdown_tool]))
+workflow.set_entry_point("agent")
+workflow.add_conditional_edges("agent", should_continue, {"tools": "tools", END: END})
+workflow.add_edge("tools", "agent")
+app = workflow.compile()
 
+
+# Document + Agent Pipeline
+async def analyze_signatures_agents(path: str, questions: List[str]) -> str:
+    """Parses the given document for signature data and analyzes it using a LangGraph agent.
+
+    Args:
+        path (str): Path to the document (PDF).
+        questions (List[str]): List of questions to ask about the detected signatures.
+
+    Returns:
+        str: Final agent response analyzing the signatures.
+    """
+
+    print("🔍 Processing document with signature detection...")
+
+    # Pass parsing options and run the tool
+    parsed_output = await document_markdown_tool.ainvoke(
+        {"path": path, "options": DocumentParserOptions(detect_signature=True)}
+    )
+
+    # Build prompt
+    prompt = build_document_analysis_prompt(parsed_output, questions)
+
+    # Run agent on prompt
+    final_state = await app.ainvoke({"messages": [HumanMessage(content=prompt)]})
+
+    return final_state["messages"][-1].content
+
+
+async def signature_detection_real_estate(path, questions):
+
+    result = await analyze_signatures_agents(path=path, questions=questions)
+
+    print("Analysis Result:\n\n", result)
+
+
+# Run an example analysis on a real estate document for signature insights
 if __name__ == "__main__":
-    # Run and test the flow
-    main()
+
+    # change to your own file path
+    path = "path/to/your/document.pdf"
+
+    analysis_questions = [
+        "How many signatures were detected in this document and who are the parties involved?",
+        "What contextual information can you extract about any signatures?",
+        "Are there any missing signatures on any pages?",
+    ]
+
+    # run the example
+    asyncio.run(signature_detection_real_estate(path, analysis_questions))
```

Comment:
If we want to move the path and questions to this part instead of in the above function, let's also update the tutorial to reflect this.